README


This archive contains the distribution Dist-Zilla-PluginBundle-ARJONES,
version 1.133170:

  L<Dist::Zilla> plugins for ARJONES

This software is copyright (c) 2013 by Andrew Jones.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
1.133170  2013-11-13 19:34:01 Europe/London
    Fix dependencies
    Set allow_dirty for Git::Check

1.131000  2013-04-10 22:14:10 Europe/London

    Add Dist::Zilla::Plugin::Test::ChangesHasContent
    Add Dist::Zilla::Plugin::ReportVersions::Tiny

1.130720  2013-03-13 21:40:55 Europe/London
    Credit contributors based on great advice by dagolden (http://www.dagolden.com/index.php/1921/how-im-using-distzilla-to-give-credit-to-contributors/)
    Add Pod::Weaver::Plugin::Encoding

1.130670  2013-03-08 14:04:40 Europe/London
    PodWeaver: Print EVENTS before ATTRIBUTES and METHODS

1.120690  2012-03-09 16:22:02 Europe/London
    Add [Test::Compile]

1.120570  2012-02-26 23:39:52 Europe/London
    Remove [Test::PodSpelling] - more trouble than its worth

1.120460  2012-02-15 12:53:29 Europe/London
    Add [Clean]

1.120240  2012-01-24 15:25:10 America/Los_Angeles
    Add [NoSmartCommentsTests]

1.120110  2012-01-11 12:10:04 Europe/London
    Fix Dist::Zilla::Plugin::NoTabsTests prereq
    Remove [EOLTests]. Too annoying

1.113460  2011-12-12 11:52:06 Europe/London
    Add [Test::Portability]
    Add [Test::PodSpelling]
    Add [Test::Pod::No404s]

1.113320  2011-11-28 13:47:14 Europe/London
    Add [NoTabsTests], [EOLTests] and [Test::Kwalitee]

1.113210  2011-11-17 23:17:29 Europe/Zurich
    Add the following missing Prereqs:
      Pod::Elemental::Transformer::List
      Test::Pod::Coverage
      Test::Pod
      Test::Perl::Critic
      Pod::Coverage::TrustPod

1.113180  2011-11-14 19:05:20 Europe/London
    Update Prereqs
    Add [PodCoverageTests]
    Add Dist::Zilla::App::Command::cover and Dist::Zilla::App::Command::perltidy as Prereqs

1.112790  2011-10-06 20:58:17 Europe/London
    Initial version.


lib/Pod/Weaver/PluginBundle/ARJONES.pm
use strict;
use warnings;

package Pod::Weaver::PluginBundle::ARJONES;
{
  $Pod::Weaver::PluginBundle::ARJONES::VERSION = '1.133170';
}

# ABSTRACT: ARJONES's default Pod::Weaver config

use Pod::Weaver::Plugin::Encoding ();
use Pod::Weaver::Section::Contributors 0.001 ();


use Pod::Weaver::Config::Assembler;
sub _exp { Pod::Weaver::Config::Assembler->expand_package( $_[0] ) }


sub mvp_bundle_config {
    my @plugins;
    push @plugins, (
        [ '@ARJONES/CorePrep', _exp('@CorePrep'), {} ],
        [ '@ARJONES/Name',     _exp('Name'),      {} ],
        [ '@ARJONES/Version',  _exp('Version'),   {} ],

        [ '@ARJONES/Prelude',  _exp('Region'),  { region_name => 'prelude' } ],
        [ '@ARJONES/Synopsis', _exp('Generic'), { header      => 'SYNOPSIS' } ],
        [
            '@ARJONES/Description', _exp('Generic'), { header => 'DESCRIPTION' }
        ],
        [ '@ARJONES/Overview', _exp('Generic'), { header => 'OVERVIEW' } ],

        [ '@ARJONES/Stability', _exp('Generic'), { header => 'STABILITY' } ],
        [ '@ARJONES/Events',    _exp('Generic'), { header => 'EVENTS' } ],
    );

    for my $plugin (
        [ 'Attributes', _exp('Collect'), { command => 'attr' } ],
        [ 'Methods',    _exp('Collect'), { command => 'method' } ],
        [ 'Functions',  _exp('Collect'), { command => 'func' } ],
      )
    {
        $plugin->[2]{header} = uc $plugin->[0];
        push @plugins, $plugin;
    }

    push @plugins,
      (
        [ '@ARJONES/Encoding',  _exp('-Encoding'), {} ],
        [ '@ARJONES/Leftovers', _exp('Leftovers'), {} ],
        [ '@ARJONES/postlude', _exp('Region'),  { region_name => 'postlude' } ],
        [ '@ARJONES/Authors',  _exp('Authors'), {} ],
        [ '@ARJONES/Contributors', _exp('Contributors'), {} ],
        [ '@ARJONES/Legal',        _exp('Legal'),        {} ],
        [ '@ARJONES/List', _exp('-Transformer'), { 'transformer' => 'List' } ],
      );

    return @plugins;
}

1;

__END__

=pod

=encoding UTF-8

=head1 NAME

Pod::Weaver::PluginBundle::ARJONES - ARJONES's default Pod::Weaver config

=head1 VERSION

version 1.133170

=head1 DESCRIPTION

This is the default Pod::Weaver config that ARJONES uses. Roughly equivalent to:

=over 4

=item *

C<@Default>

=item *

C<-Transformer> with L<Pod::Elemental::Transformer::List>

=back

Heavily based on L<Pod::Weaver::PluginBundle::RJBS>.

=for Pod::Coverage mvp_bundle_config

=head1 AUTHOR

Andrew Jones <andrew@arjones.co.uk>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2013 by Andrew Jones.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "L<Dist::Zilla> plugins for ARJONES",
  "AUTHOR" => "Andrew Jones <andrew\@arjones.co.uk>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-ARJONES",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dist::Zilla::PluginBundle::ARJONES",
  "PREREQ_PM" => {
    "Dist::Zilla" => "2.100922",
    "Dist::Zilla::App::Command::cover" => 0,
    "Dist::Zilla::App::Command::perltidy" => 0,
    "Dist::Zilla::Plugin::Clean" => 0,
    "Dist::Zilla::Plugin::ContributorsFromGit" => 0,
    "Dist::Zilla::Plugin::GithubMeta" => 0,
    "Dist::Zilla::Plugin::NoSmartCommentsTests" => 0,
    "Dist::Zilla::Plugin::NoTabsTests" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::ReportVersions::Tiny" => 0,
    "Dist::Zilla::Plugin::Test::ChangesHasContent" => 0,
    "Dist::Zilla::Plugin::Test::Compile" => 0,
    "Dist::Zilla::Plugin::Test::Kwalitee" => 0,
    "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
    "Dist::Zilla::Plugin::Test::Pod::No404s" => 0,
    "Dist::Zilla::Plugin::Test::Portability" => 0,
    "Dist::Zilla::PluginBundle::Basic" => 0,
    "Dist::Zilla::PluginBundle::Git" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Moose" => 0,
    "Moose::Autobox" => 0,
    "Pod::Coverage::TrustPod" => 0,
    "Pod::Elemental::Transformer::List" => 0,
    "Pod::Weaver::Config::Assembler" => 0,
    "Pod::Weaver::Plugin::Encoding" => 0,
    "Pod::Weaver::Section::Contributors" => "0.001",
    "Test::NoSmartComments" => 0,
    "Test::NoTabs" => "1.3",
    "Test::Perl::Critic" => 0,
    "Test::Pod" => "1.41",
    "Test::Pod::Coverage" => "1.08",
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.88"
  },
  "VERSION" => "1.133170",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla" => "2.100922",
  "Dist::Zilla::App::Command::cover" => 0,
  "Dist::Zilla::App::Command::perltidy" => 0,
  "Dist::Zilla::Plugin::Clean" => 0,
  "Dist::Zilla::Plugin::ContributorsFromGit" => 0,
  "Dist::Zilla::Plugin::GithubMeta" => 0,
  "Dist::Zilla::Plugin::NoSmartCommentsTests" => 0,
  "Dist::Zilla::Plugin::NoTabsTests" => 0,
  "Dist::Zilla::Plugin::PodWeaver" => 0,
  "Dist::Zilla::Plugin::ReportVersions::Tiny" => 0,
  "Dist::Zilla::Plugin::Test::ChangesHasContent" => 0,
  "Dist::Zilla::Plugin::Test::Compile" => 0,
  "Dist::Zilla::Plugin::Test::Kwalitee" => 0,
  "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
  "Dist::Zilla::Plugin::Test::Pod::No404s" => 0,
  "Dist::Zilla::Plugin::Test::Portability" => 0,
  "Dist::Zilla::PluginBundle::Basic" => 0,
  "Dist::Zilla::PluginBundle::Git" => 0,
  "Dist::Zilla::Role::PluginBundle::Easy" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Moose" => 0,
  "Moose::Autobox" => 0,
  "Pod::Coverage::TrustPod" => 0,
  "Pod::Elemental::Transformer::List" => 0,
  "Pod::Weaver::Config::Assembler" => 0,
  "Pod::Weaver::Plugin::Encoding" => 0,
  "Pod::Weaver::Section::Contributors" => "0.001",
  "Test::More" => "0.88",
  "Test::NoSmartComments" => 0,
  "Test::NoTabs" => "1.3",
  "Test::Perl::Critic" => 0,
  "Test::Pod" => "1.41",
  "Test::Pod::Coverage" => "1.08",
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



